apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LoggingConfiguration.java

```diff
@@ -298,34 +294,11 @@ public void assertValid(Boolean value) {
         .buildWithDefault(LogFormat.PLAIN_TEXT);
 
     public static void init(List<ConfigurationSource> sources, String ephemeralId) {
-        // The initialization of log4j may produce errors if the traced application uses log4j settings (for
-        // example - through file in the classpath or System properties) that configures specific properties for
-        // loading classes by name. Since we shade our usage of log4j, such non-shaded classes may not (and should not)
-        // be found on the classpath.
-        // All handled Exceptions should not prevent us from using log4j further, as the system falls back to a default
-        // which we expect anyway. We take a calculated risk of ignoring such errors only through initialization time,
-        // assuming that errors that will make the logging system non-usable won't be handled.
-        String initialListenersLevel = System.setProperty(INITIAL_LISTENERS_LEVEL, "OFF");
-        String initialStatusLoggerLevel = System.setProperty(INITIAL_STATUS_LOGGER_LEVEL, "OFF");
-        String defaultListenerLevel = System.setProperty(DEFAULT_LISTENER_LEVEL, "OFF");
-        try {
-            Configurator.initialize(new Log4j2ConfigurationFactory(sources, ephemeralId).getConfiguration());
-        } catch (Throwable throwable) {
-            System.err.println("[elastic-apm-agent] ERROR Failure during initialization of agent's log4j system: " + throwable.getMessage());
-        } finally {
-            restoreSystemProperty(INITIAL_LISTENERS_LEVEL, initialListenersLevel);
-            restoreSystemProperty(INITIAL_STATUS_LOGGER_LEVEL, initialStatusLoggerLevel);
-            restoreSystemProperty(DEFAULT_LISTENER_LEVEL, defaultListenerLevel);
-            StatusLogger.getLogger().setLevel(Level.ERROR);
-        }
-    }
-
-    private static void restoreSystemProperty(String key, @Nullable String originalValue) {
-        if (originalValue != null) {
-            System.setProperty(key, originalValue);
-        } else {
-            System.clearProperty(key);
-        }
+        // org.apache.logging.log4j.core.config.ConfigurationFactory is a singleton that allows overriding its instance
+        // through setConfigurationFactory. This API is not considered thread safe, but since we do it so early on when
+        // there is only the main thread it should be OK. Once we override the default factory instance, any logger
+        // created thereafter will be configured through our custom factory.
```

Comment:
```suggestion
        // created thereafter will be configured through our custom factory.
        // This avoids that loggers in plugin class loaders get their own context based on the thread context class loader,
        // which would initialize the loggers based on the properties/xml file that the target application ships with
```

Reply:
I'll add something in this sprit, thanks.
The default context selector is using the caller class loader, not the thread context CL and the problem is either when there is existing log4j config in the classpath or there isn't (then we get an error of non-existing config)